superset/models/dashboard.py

```diff
@@ -312,10 +312,10 @@ def clear_cache_for_datasource(cls, datasource_id: int) -> None:
             [dashboard_slices.c.dashboard_id], distinct=True,
         ).select_from(
             join(
-                Slice,
                 dashboard_slices,
-                Slice.id == dashboard_slices.c.slice_id,
-                Slice.datasource_id == datasource_id,
+                Slice,
+                (Slice.id == dashboard_slices.c.slice_id)
+                & (Slice.datasource_id == datasource_id),
```

Comment:
nit: I think this could be expressed as `and_(Slice.id == dashboard_slices.c.slice_id, Slice.datasource_id == datasource_id)`

Reply:
I don't know, I kind of like the bit operator better..